pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -424,7 +424,9 @@ func realScheduleMonitor(c *Cluster, mon *monConfig) (SchedulingResult, error) {
 
 	// setup affinity settings for pod scheduling
 	p := cephv1.GetMonPlacement(c.spec.Placement)
-	c.setPodPlacement(&d.Spec.Template.Spec, p, nil)
+	if !c.spec.Mon.AllowMultiplePerNode {
```

Comment:
Please extract this and make it its own PR.

Reply:
yeah done https://github.com/rook/rook/pull/4741